src/assign.c

```diff
@@ -230,57 +245,52 @@ SEXP alloccol(SEXP dt, R_len_t n, Rboolean verbose)
         // otherwise the finalizer can't clear up the Large Vector heap
   return(dt);
 }
+*/
 
-int checkOverAlloc(SEXP x)
+int allocColOpt()
 {
-  if (isNull(x))
-    error(_("Has getOption('datatable.alloccol') somehow become unset? It should be a number, by default 1024."));
-  if (!isInteger(x) && !isReal(x))
-    error(_("getOption('datatable.alloccol') should be a number, by default 1024. But its type is '%s'."), type2char(TYPEOF(x)));
-  if (LENGTH(x) != 1)
-    error(_("getOption('datatable.alloc') is a numeric vector ok but its length is %d. Its length should be 1."), LENGTH(x));
+  SEXP x = GetOption(sym_alloccol, R_NilValue);
+  if (length(x)!=1 || (!isInteger(x) && !isReal(x)))
+    error(_("%s should be a single number but it is length %d and type '%s'."), "getOption('datatable.alloccol')", length(x), type2char(TYPEOF(x)));
   int ans = isInteger(x) ? INTEGER(x)[0] : (int)REAL(x)[0];
   if (ans<0)
-    error(_("getOption('datatable.alloc')==%d.  It must be >=0 and not NA."), ans);
+    error(_("%s is %d. It must be >=0 and not NA."), "getOption('datatable.alloccol')", ans);
   return ans;
 }
 
-SEXP alloccolwrapper(SEXP dt, SEXP overAllocArg, SEXP verbose) {
-  if (!IS_TRUE_OR_FALSE(verbose))
+SEXP alloccolwrapper(SEXP dt, SEXP nSpareArg, SEXP verboseArg) {
+  if (TYPEOF(dt) != VECSXP)
+    error(_("dt passed to alloccol isn't type VECSXP"));
+  if (!IS_TRUE_OR_FALSE(verboseArg))
     error(_("%s must be TRUE or FALSE"), "verbose");
-  int overAlloc = checkOverAlloc(overAllocArg);
-  SEXP ans = PROTECT(alloccol(dt, length(dt)+overAlloc, LOGICAL(verbose)[0]));
-
-  for(R_len_t i = 0; i < LENGTH(ans); i++) {
-    // clear names; also excluded by copyMostAttrib(). Primarily for data.table and as.data.table, but added here centrally (see #103).
-    setAttrib(VECTOR_ELT(ans, i), R_NamesSymbol, R_NilValue);
-
-    // But don't clear dim and dimnames. Because as from 1.12.4 we keep the matrix column as-is and ask user to use as.data.table to
-    // unpack matrix columns when they really need to; test 2089.2
-    // setAttrib(VECTOR_ELT(ans, i), R_DimSymbol, R_NilValue);
-    // setAttrib(VECTOR_ELT(ans, i), R_DimNamesSymbol, R_NilValue);
-  }
+  const bool verbose = LOGICAL(verboseArg)[0];
+  if (length(nSpareArg)!=1 || (!isInteger(nSpareArg) && !isReal(nSpareArg)))
+    error(_("%s should be a single number but it is length %d and type '%s'"), "getOption('datatable.alloccol') (or an argument with this default)", length(nSpareArg), type2char(TYPEOF(nSpareArg)));
+  int nSpare = isInteger(nSpareArg) ? INTEGER(nSpareArg)[0] : (int)REAL(nSpareArg)[0];
```

Comment:
won't work for `bit64` integer such as in `alloc.col(DT, as.integer64(100))` (quite edgy though)

Reply:
Indeed. We probably have this problem throughout many arguments in many functions then? On the one hand I agree it's edgy but now you've raised it, ugh, yep. Maybe we need a new helper to deal with this and sweep the whole codebase in a separate PR for int/real arguments that could be int64.

Reply:
coerceAs helper was added to handle this kind of coercion

Reply:
I can see `coerceAs` defined in `wrappers.R` but it doesn't seem in use at R level yet. Yes it would be good to start using it.

Reply:
AFAIR I added R wrapper for easier tests coverage.